autonomic-plugin-common/src/main/java/br/com/autonomiccs/autonomic/plugin/common/daos/AutonomiccsSystemVmTemplateJdbcDao.java

```diff
@@ -43,18 +46,30 @@ public boolean isTemplateRegistered(String templateName) {
         return executeTemplateQueryAndRetrieveBoolean(templateName, sqlIsTemplateRegistered);
     }
 
-    private boolean executeTemplateQueryAndRetrieveBoolean(String templateName, String sql) {
+    /**
+     * Given a template name and a sql query, it executes the sql query with the template name; the
+     * query returns the number of registers for the given query; if the number of registers is
+     * equals to zero (0) it returns false, if it is equals to one (1) it returns true; in case of
+     * more registers than one (1) it throws the {@link CloudRuntimeException}.
+     */
+    protected boolean executeTemplateQueryAndRetrieveBoolean(String templateName, String sql) {
         Integer numberOfRegister = getJdbcTemplate().queryForObject(sql, new Object[] {templateName}, Integer.class);
         if (numberOfRegister > 1) {
             throw new CloudRuntimeException(String.format("More than one template with name [%s]", templateName));
         }
         return BooleanUtils.toBoolean(numberOfRegister);
     }
 
+    /**
+     * It returns 'true' if the template is registered and ready.
```

Comment:
I think you should complete the description. "... registered and ready to be used"